pkg/core/native/native_neo.go

```diff
@@ -447,33 +446,12 @@ func (n *NEO) PostPersist(ic *interop.Context) error {
 	return nil
 }
 
-func (n *NEO) getGASPerVote(d *dao.Simple, key []byte, indexes []uint32) []big.Int {
-	sort.Slice(indexes, func(i, j int) bool {
-		return indexes[i] < indexes[j]
-	})
-	start := make([]byte, 4)
-	binary.BigEndian.PutUint32(start, indexes[len(indexes)-1])
-
-	need := len(indexes)
-	var reward = make([]big.Int, need)
-	collected := 0
-	d.Seek(n.ID, storage.SeekRange{
-		Prefix:    key,
-		Start:     start,
-		Backwards: true,
-	}, func(k, v []byte) bool {
-		if len(k) == 4 {
-			num := binary.BigEndian.Uint32(k)
-			for i, ind := range indexes {
-				if reward[i].Sign() == 0 && num <= ind {
-					reward[i] = *bigint.FromBytes(v)
-					collected++
-				}
-			}
-		}
-		return collected < need
-	})
-	return reward
+func (n *NEO) getLatestGASPerVote(d *dao.Simple, key []byte) big.Int {
+	item := d.GetStorageItem(n.ID, key)
```

Comment:
The beauty of this scheme wrt NeoGo is that you have this value in the cache already (see `gasPerVoteCache`) most of the time. So you can reuse it and it'll be even faster than now.

Reply:
Done